extensions/wikia/Chat2/ChatBanListSpecial_helper.php

```diff
@@ -0,0 +1,269 @@
+			$this->load();
+		}
+	}
+
+	function load() {
+		$this->setLimit();
+		$this->setOffset();
+		$this->setOrder();
+	}
+
+	function setUserName( $username = '' ) { $this->userName = $username; }
+
+	function setLimit( $limit = Listusers::DEF_LIMIT ) { $this->limit = $limit; }
+
+	function setOffset( $offset = 0 ) { $this->offset = $offset; }
+
+	function setOrder( $order = '' ) {
+		if ( empty( $order ) ) {
+			// default order
+			$order = 'timestamp:desc';
+		}
+
+		$validSortDirections = [
+			'asc',
+			'desc',
+		];
+
+		list ( $orderName, $orderDesc ) = explode( ":", $order );
+		if ( isset( $this->orderOptions[ $orderName ] ) && in_array( $orderDesc, $validSortDirections ) ) {
+			$this->order = sprintf( $this->orderOptions[ $orderName ], $orderDesc );
+		}
+	}
+
+	function getUserName() { return $this->userName; }
+
+	function getLimit() { return $this->limit; }
+
+	function getOffset() { return $this->offset; }
+
+	function getOrder() { return $this->order; }
+
+	public function loadData() {
+		global $wgMemc, $wgLang;
+		wfProfileIn( __METHOD__ );
+
+		/* initial values for result */
+		$data = [
+			'cnt'      => 0,
+			'sColumns' => implode( ",", array_keys( $this->orderOptions ) ),
+			'data'     => [ ],
+		];
+
+		$subMemkey = [
+			'O' . $this->offset,
+			'O' . $this->order,
+			'L' . $this->limit,
+			'U' . $this->userName,
+		];
+
+		$memkey = wfForeignMemcKey( $this->cityId, null, "cbdata", md5( implode( ', ', $subMemkey ) ) );
+		$cached = $wgMemc->get( $memkey );
+
+		if ( empty( $cached ) ) {
+			/* db handle */
+			$dbs = wfGetDB( DB_SLAVE, [ ], $this->db );
+
+			/* initial conditions for SQL query */
+			$where = [
+				'cbu_wiki_id' => $this->cityId,
+				"cbu_user_id != ''",
+				"end_date > '" . wfTimestamp( TS_MW ) . "'",
+			];
+
+			/* filter: user name */
+			$userId = User::IdFromName( $this->userName );
+
+			if ( $userId !== null ) {
+				$where[] = " cbu_user_id = " . $dbs->addQuotes( $userId );
+			}
+
+			/* number of records */
+			$row = $dbs->selectRow(
+				$this->table,
+				[ 'count(0) as cnt' ],
+				$where,
+				__METHOD__
+			);
+			if ( is_object( $row ) ) {
+				$data['cnt'] = $row->cnt;
+			}
+
+			if ( $data['cnt'] > 0 ) {
+				/* select records */
+				$oRes = $dbs->select(
+					[
+						$this->table,
+					],
+					[
+						'start_date',
+						'cbu_user_id',
+						'end_date',
+						'cbu_admin_user_id',
+						'reason',
+					],
+					$where,
+					__METHOD__,
+					[
+						'ORDER BY' => $this->order,
+						'LIMIT'    => $this->limit,
+						'OFFSET'   => intval( $this->offset ),
+					]
+				);
+
+				$data['data'] = [ ];
+				while ( $row = $dbs->fetchObject( $oRes ) ) {
+
+					$user = User::newFromId( $row->cbu_user_id );
+					$admin = User::newFromId( $row->cbu_admin_user_id );
+
+					$data['data'][] = [
+						'timestamp'    => $wgLang->timeanddate( $row->start_date, true ),
+						'user'         => $this->skin->link( $user->getUserPage(), $user->getName() ),
```

Comment:
Using a `Skin` instance for linking is deprecated. Please use the static methods of `Linker` class such as `Linker::link` - https://github.com/Wikia/app/blob/dev/includes/Linker.php